enhancements/etcd/assisted-install.md

```diff
@@ -0,0 +1,221 @@
+1. Allows a situation where we can get stuck unable to re-achieve quorum, resulting in cluster-death - unsafe, non-HA, non-production, unstable
+1. The combination of all these things makes the situation unsupportable. 
+
+To allow toleration of this check we provided an unsupportedConfigOverrides
+`useUnsupportedUnsafeNonHANonProductionUnstableEtcd`. As assisted installer is
+now moving to GA a supported toleration is required.
+
+## Goals
+
+1.  Provide a supported solution for assisted installer that will allow
+    `cluster-etcd-operator` to tolerate 2 master replicas as install complete.
+
+## Non-Goals
+
+1.  Support the current unsupportedConfigOverrides
+1.  Add support for less than 3 master node clusters
+
+## Currently Supported Functionality 
+Currently `cluster-etcd-operator` and OpenShift only supports clusters with
+three master nodes as install complete.
+
+### EnvVarController
+Currently the `EnvVarController` ensures that all of the data required for the
+etcd static pods exists. A condition exists within the `getEtcdEnvVars`[1]  function which checks if we have the expected master node count or if
+`useUnsupportedUnsafeNonHANonProductionUnstableEtcd` is populated otherwise
+return error which results in the controller being degraded and no static-pod
+spec installed on the master node for etcd.
+
+[1] https://github.com/openshift/cluster-etcd-operator/blob/be2b8d97ca048baecb051d288971a9f6db411457/pkg/etcdenvvar/etcd_env.go#L71
+
+## Proposed changes
+The following changes are proposed.
+
+In order for the operator to tolerate an unsafe install configuration all
+efforts must be taken to minimize the risk of tolerating user defined unsafe
+cluster. For example to save on costs user chooses 2 master replicas.
+
+1.  Observe master replica count from install-config.
+1.  Observe install status and only tolerate less than 3 masters until install
+    complete. After install status is complete the operator will go into a
+degraded state until the cluster has 3 or more master nodes.
+1.  Observe infrastructure and only tolerate `BareMetal` type. Assumption being
+    that assisted installer will GA with support for only `BareMetal` (true?)
+
+### cluster-config-v1 configmap
+The expected master replica count provided by the `install-config` is a very
+important value for this design. This value will be used as the desired state
+for nodes during install. By using this value we can protect against allowing a cluster which
+expects 3 master nodes to be install complete if one of those nodes fails to
+pull ignition for example.
+
+cluster-config-v1 configmap with install-config embedded
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-config-v1
+  namespace: kube-system
+data:
+  install-config: |
+    apiVersion: v1
+    baseDomain: group-b.devcluster.openshift.com
+    controlPlane:
+      architecture: amd64
+      hyperthreading: Enabled
+      name: master
+      platform:
+        [..]
+      replicas: 3
```

Comment:
@yevgeny-shnaidman we'll be able to update this manifest before generating the ignitions, right?

Reply:
@eranco74

Reply:
if it is in bootstrap or master ignition we can do it

Reply:
Why would we?
Why not have the right number in the install config?

Reply:
this was just an example config in your case you would set it to 2. the change to the install-config was not intended to be replica count. But a possible addition of 'installType'  I will update it to be more clear.

Reply:
Actually now I understand that 3 is the actual count that you will use? @romfreiman please confirm.

Reply:
yep.